---
title: "Integrate UGA leaf pipseq and multiome snRNA from "
date: 4-June-2025
author: "Anne Pavia"
output:
  html_document:
    df_print: paged
---
paste in termial:  setwd("/scratch/ac05869/10X_Multiome/KRT_leaf/seurat_out/wd")
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_knit$set(root.dir = '/scratch/ac05869/10X_Multiome/KRT_leaf/seurat_out/wd')
setwd("/scratch/ac05869/10X_Multiome/KRT_leaf/seurat_out/wd")
```

# packages 
```{r}
library(tidyverse)
library(Seurat)
library(readxl)
library(RColorBrewer)
library(rcartocolor)
library(viridis)
library(patchwork)
library(showtext)
library(ggpubr)
showtext_auto(enable = F) #This makes it so you can see the text in the graphs, but it saves the images weird to turn to FALSE before ggsave
```

# Pipseq RNA: uga kratom leaf nuclei
##Load RDS Object 
```{r}
AA_clean <- readRDS(file = "/scratch/ac05869/KRT_AA_AB_sc/seurat.out/KRT_AA_clean_normalized.rds")
AB_clean <- readRDS(file = "/scratch/ac05869/KRT_AA_AB_sc/seurat.out/KRT_AB_clean_normalized.rds")

```

#Multiome RNA: uga kratom leaf nuclei
##Load RDS Object
```{r}
AE_clean <- readRDS(file = "../Data/AE_clean_normalized.rds")
AF_clean <- readRDS(file = "../Data/AF_clean_normalized.rds")
```

# Integration via reciprocal PCA (RPCA)
## Notes
This method is what was suggested in Suerat v4 as of Oct 2023. In their new vignette for Seurat v5 the methodology looks to be updated. This may warrant investigation prior to running the below! 
I still recommend RPCA, see the linked vginette below for reasons why.
https://satijalab.org/seurat/articles/integration_rpca
We therefore,recommend RPCA during integrative analysis where: * A substantial fraction of cells in one dataset have no matching type in the other

##Data Prep
```{r}
AA_clean
AB_clean
AE_clean
AF_clean
AA_clean$source <- "pipseq"
AB_clean$source <- "pipseq"
AE_clean$source <- "multiome"
AF_clean$source <- "multiome"

leaf.list <- list(AA_clean, AB_clean, AE_clean, AF_clean)

# normalize and identify variable features for each dataset independently
leaf.list <- lapply(X = leaf.list, FUN = function(x) {
    x <- NormalizeData(x)
    x <- FindVariableFeatures(x, selection.method = "vst", nfeatures = 3000)
})

# select features that are repeatedly variable across datasets for integration run PCA on each dataset using these features
features <- SelectIntegrationFeatures(object.list = leaf.list, nfeatures = 3000)
leaf.list <- lapply(X = leaf.list, FUN = function(x) {
    x <- ScaleData(x, features = features, verbose = FALSE)
    x <- RunPCA(x, features = features, verbose = FALSE)
})
```

## Integration 
```{r}
leaf.anchors <- FindIntegrationAnchors(object.list = leaf.list, anchor.features = features, reduction = "rpca")

# this command creates an 'integrated' data assay
KRT_RNA <- IntegrateData(anchorset = leaf.anchors)

# specify that we will perform downstream analysis on the corrected data note that the
# original unmodified data still resides in the 'RNA' assay
DefaultAssay(KRT_RNA) <- "integrated"
```

## Stats
```{r}
KRT_RNA

summary(KRT_RNA$nFeature_RNA)
summary(KRT_RNA$nCount_RNA)
```

## Scale the Data
```{r}
KRT_RNA.genes <- rownames(KRT_RNA)
KRT_RNA <- ScaleData(KRT_RNA, features = KRT_RNA.genes)
```

## PCA
```{r}
KRT_RNA$Run <- KRT_RNA@meta.data$orig.ident
KRT_RNA <- RunPCA(KRT_RNA, npcs = 75, features = VariableFeatures(object = KRT_RNA), verbose = FALSE) # run PCA
ElbowPlot(KRT_RNA, ndims = 75)
```

## Call Clusters
```{r}
KRT_RNA <- FindNeighbors(KRT_RNA, dims = 1:50) #adjust dims based on elbow plot above
KRT_RNA <- FindClusters(KRT_RNA, resolution = 0.5) #adjustable resolution value. Check back and forth until it looks good
KRT_RNA <- RunUMAP(KRT_RNA, dims = 1:50, reduction.name = "umap.rna", reduction.key = "rnaUMAP_")
```

## Make UMAP Plots
```{r}
UMAP1 <- DimPlot(KRT_RNA, reduction = "umap", label = T, label.size = 5, repel = T, split.by = "source") +
  theme_void() +
  theme(
    text = element_text(size = 14, color = "black", face = "bold"),
    legend.position = "none",
    title = element_text(size = 14)
  ) +
  ggtitle("Grouped by Source\n")
UMAP1
#check to see that there are the same number of cluster in each data source
ggsave("../Integrated_RNA_Results/KRT_RNA_integrated_umap_by_source.svg", height = 8, width = 12, bg = "white")
ggsave("../Integrated_RNA_Results/KRT_RNA_integrated_umap_by_source.png", height = 8, width = 12, bg = "white")
```

```{r}
UMAP2 <- DimPlot(KRT_RNA, reduction = "umap", label = T, label.size = 5, repel = T) +
#  theme_void() +
  labs(
    title = "Integrated Leaf sn-RNA") +
  theme(
    text = element_text(size = 14, color = "black", face = "bold"),
    legend.position = "none",
    title = element_text(size = 14)
  )
UMAP2
ggsave("../Integrated_RNA_Results/KRT_RNA_integrated_umap.svg", height = 10, width = 12, bg = "white")
ggsave("../Integrated_RNA_Results/KRT_RNA_integrated_umap.png", height = 10, width = 12, bg = "white")
```

```{r}
DimPlot(KRT_RNA, group.by = "source", reduction = "umap") 
UMAP3 <- DimPlot(KRT_RNA, split.by = "source", group.by = "Run", reduction = "umap") +
  theme_void() +
  labs(
    title = "Split by Rep") +
  theme(
    text = element_text(size = 14, color = "black", face = "bold"),
    title = element_text(size = 14)
  )
UMAP3
ggsave("../Integrated_RNA_Results/KRT_RNA_integrated_umap_by_rep.svg", height = 8, width = 12, bg = "white")
ggsave("../Integrated_RNA_Results/KRT_RNA_integrated_umap_by_rep.png", height = 8, width = 12, bg = "white")
```




## Stats - Integrated Clusters
```{r}
#Cells in each cluster
table(Idents(KRT_RNA))

# Cells in each replicate
table(KRT_RNA$orig.ident)

#How does cluster membership vary by source?
table(Idents(KRT_RNA), KRT_RNA$source)

#How does cluster membership vary by source?
table(Idents(KRT_RNA), KRT_RNA$Run)
```
# Create Matrices
## Pull annotation
```{r}
anno_mitr <- read.delim(file = "/scratch/ac05869/nih/kratom/mitr_v1_anno/mitr_v1.functional_annotation.txt", header = FALSE, sep = "\t", col.names = c("gene_ID", "functional_annotation"))
anno_mitr <- anno_mitr %>%
  mutate(gene_ID = str_replace(gene_ID, "_", "-"))
head(anno_mitr)
dim(anno_mitr) #172819


repr_gene_models <- read_delim("../../genome/mitr_v1.working_models.repr.gff3",
                               delim = "\t", col_names = F)
dim(repr_gene_models) #1152702

repr_loci <- repr_gene_models %>% 
  dplyr::filter(X3 == "mRNA") %>% 
  dplyr::select(X1, X4, X5, X7, X9) %>% 
  separate(X9, c("ID", "Name"), sep = ";") %>% 
  dplyr::select(-Name) %>% 
  separate(ID, c("temp", "gene_ID"), sep = "=") %>% 
  dplyr::select(-temp) %>% 
  mutate(gene_ID = str_replace(gene_ID, "_", "-")) %>% 
  dplyr::rename(
    Chr = X1,
    start = X4,
    end = X5,
    strand = X7
  )

head(repr_loci)
dim(repr_loci) #103160

anno <- anno_mitr %>%
  right_join(repr_loci, by = "gene_ID", relationship = "one-to-one")%>% 
  mutate(gene_ID = str_replace(gene_ID, "\\.\\d+$", "")) 
dim(anno) #103160
head(anno)
```
## Pull MIA genes 
```{r}
#Need to create a file that has your marker genes and some metadata (cell types, etc.). Change all underscores to either dashes or periods

pathway_genes <- read.table("../Data/pathway_orthologs_kratom.txt", header = TRUE, sep = "\t")
head(pathway_genes)
pathway_genes <- pathway_genes %>%
  mutate(gene_ID = pathway_genes$kratom.id) %>%
  mutate(gene_ID = str_replace(gene_ID, "_", "-")) %>%
  mutate(gene_ID = str_replace(gene_ID, "\\.\\d+$", "")) %>% 
  mutate(tag = paste(gene.abbreviation, ":", kratom.id)) %>%
  left_join(anno, by = "gene_ID")

rev_pathway_genes <- pathway_genes[order(nrow(pathway_genes):1),]
head(rev_pathway_genes)
write_excel_csv(pathway_genes, "../Integrated_RNA_Results/leaf_MIA_orthologs.csv")

#Pull top blast goi from allwin
goi <- read.table("../Data/allwin_goi_top_blast.txt", header = TRUE, sep = "\t")
head(goi)
goi <- goi %>%
  mutate(gene_ID = goi$kratom.id) %>%
  mutate(gene_ID = str_replace(gene_ID, "_", "-")) %>%
  mutate(gene_ID = str_replace(gene_ID, "\\.\\d+$", "")) %>% 
  mutate(tag = paste(gene.abbreviation, ":", gene_ID)) %>%
  mutate(pathway = str_detect())
  left_join(anno, by = "gene_ID")
head(goi)
rev_goi <- goi[order(nrow(goi):1),]
head(rev_goi)
write_excel_csv(goi, "../Integrated_RNA_Results/KRT_MIA_goi_blast.csv")


all_pathway <- goi %>%
  bind_rows(pathway_genes) %>%
  mutate(gene_ID = str_replace(kratom.id, "_", "-")) %>%
  mutate(gene_ID = str_replace(gene_ID, "\\.\\d+$", ""))
  
  
```
## Find Average Expression
```{r}
DefaultAssay(KRT_RNA) <- "RNA"

# replace spaces with underscores '_' so ggplot2 doesn't fail
orig.levels <- levels(KRT_RNA)
Idents(KRT_RNA) <- gsub(pattern = " ", replacement = "_", x = Idents(KRT_RNA))
orig.levels <- gsub(pattern = " ", replacement = "_", x = orig.levels)
levels(KRT_RNA) <- orig.levels

cluster.averages <- AverageExpression(KRT_RNA, return.seurat = TRUE, )
cluster.averages
cluster_avg.data <- as.matrix(GetAssayData(cluster.averages, layer = "data")[, ])
saveRDS(cluster_avg.data, file = "../Integrated_RNA_Data/KRT_RNA_cluster_expression.rds")

cluster_avg.data <- readRDS(file = "../Integrated_RNA_Data/KRT_RNA_cluster_expression.rds")

cluster_avg.data_anno <- cluster_avg.data %>%
  as.data.frame() %>%
  rownames_to_column(var = "gene_ID") %>%
  left_join(anno, by = "gene_ID", relationship = "one-to-one")
head(cluster_avg.data_anno)
  
write_delim(cluster_avg.data_anno, file = "../Integrated_RNA_Results/KRT_RNA_integrated_cluster_av_expression.txt", delim = "\t")

cluster_avg.long <- cluster_avg.data %>%
  as.data.frame() %>%
  rownames_to_column(var = "gene_ID") %>%
  pivot_longer(cols = as.character(c(0:21)), names_to = "cluster", values_to = "avg.exp") 
cluster_avg.long$cluster <- cluster_avg.long$cluster %>%
  as.factor()
head(cluster_avg.long)
write_delim(cluster_avg.long, file = "../Integrated_RNA_Results/KRT_RNA_integrated_cluster_av_expression_long.txt", delim = "\t")
```


## Find Markers
```{r}
DefaultAssay(KRT_RNA) <- "RNA"

KRT_RNA = JoinLayers(KRT_RNA)
head(KRT_RNA)

KRT_RNA_markers <- FindAllMarkers(KRT_RNA, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25) #min.pct is the minimum fraction that cells cluster needs to be expressing that gene for it to count. logfc.threshhold is the log-fold change in expression between clusters expressing that gene
saveRDS(KRT_RNA_markers, file = "../Integrated_RNA_Data/KRT_RNA_markers.rds")

#KRT_RNA_markers<- readRDS(file = "../Integrated_RNA_Data/KRT_RNA_markers.rds")
KRT_RNA_markers
dim(KRT_RNA_markers)
KRT_RNA_markers_anno <- KRT_RNA_markers %>%
  dplyr::rename(gene_ID = gene) %>%
  rownames_to_column(var = "rownames") %>%
  select(!rownames) %>%
  left_join(anno, by = "gene_ID") %>%
  left_join(cluster_avg.long, by = c("gene_ID", "cluster"), relationship = "one-to-one") %>%
  select(gene_ID, cluster, p_val, p_val_adj, avg_log2FC, pct.1, pct.2, avg.exp, functional_annotation, start,end,strand)
KRT_RNA_markers_anno
dim(KRT_RNA_markers_anno)
write_delim(KRT_RNA_markers_anno, "../Integrated_RNA_Results/KRT_RNA_integrated_markers.txt", delim = "\t")
```

## Save cleaned integrated RDS Object
```{r}
saveRDS(KRT_RNA, file = "../Integrated_RNA_Data/KRT_RNA_integrated_5june25.rds")
#KRT_RNA <- readRDS("../Integrated_RNA_Data/KRT_RNA_integrated_5june25.rds")
```
