---
title: "MIT_AH_AI_cluster_annotation"
author: "Anne Pavia"
date: "2025-01-27"
output: html_document
---
# Notes:
This is the same as "31jan25_MIT_AH_AO...final.Rmd", except I used the top several hits between Dionne's marker genes and de-duplicated them before choosing only the top hit
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# packages 
```{r}
library(tidyverse)
library(Seurat)
library(readxl)
library(RColorBrewer)
library(rcartocolor)
library(viridis)
library(patchwork)
```

# Pull Seurat data
```{r}
leaf.combined <- readRDS("/scratch/ac05869/nih/kratom/seurat_analysis/mpi_kratom_leaf_mitr.v1/MIT_AH_AI_cleaned_integrated_seurat_14jan25.rds")
DefaultAssay(leaf.combined) <- "RNA"
leaf.combined = JoinLayers(leaf.combined)
leaf.combined.markers <- FindAllMarkers(leaf.combined, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25) %>% group_by(cluster) #min.pct is the minimum fraction that cells cluster needs to be expressing that gene for it to count. logfc.threshhold is the log-fold change in expression between clusters expressing that gene
```

## Pull annotation and repr_loci
```{r}
repr_gene_models <- read_delim("/scratch/ac05869/nih/kratom/mitr_v1_anno/mitr_v1.hc_gene_models.repr.gff3",
                               delim = "\t", col_names = F)

repr_loci <- repr_gene_models %>% 
  dplyr::filter(X3 == "mRNA") %>% 
  dplyr::select(X1, X4, X5, X7, X9) %>% 
  separate(X9, c("ID", "Name"), sep = ";") %>% 
  dplyr::select(-Name) %>% 
  separate(ID, c("temp", "gene_ID"), sep = "=") %>% 
  dplyr::select(-temp) %>% 
  mutate(LocusID = str_sub(gene_ID, start = 1, end = 13)) %>% 
  mutate(gene_ID = str_replace(gene_ID, "_", "-")) %>% 
  mutate(LocusID = str_replace(LocusID, "_", "-")) %>% 
  dplyr::rename(
    Chr = X1,
    start = X4,
    end = X5,
    strand = X7
  )
head(repr_loci)

anno <- read.delim(file = "/scratch/ac05869/nih/kratom/mitr_v1_anno/mitr_v1.functional_annotation.txt", header = FALSE, sep = "\t", col.names = c("gene_ID", "functional_annotation")) %>% 
  #mutate(gene_ID = str_replace(gene_ID, "_", "-")) %>%
  mutate(gene_ID = str_replace(gene_ID, "\\.\\d+$", ""))
head(anno)
dim(anno)

head(anno_repr_loci <- inner_join(anno, repr_loci, by= "gene_ID"))
```

# Pull marker genes and blast hits
```{r}
markers <- read_delim("/scratch/ac05869/nih/marker_genes_dionne/From_Dionne/dionne_marker_genes_list.tsv", col_types = cols())
head(markers)
dim(markers)

kratom_genes <- read_delim("/scratch/ac05869/nih/marker_genes_dionne/blast/bestHits_blastn_hits.34761323.tsv", col_names = c("Ath", "gene_ID", "pident", "length", "mismatch", "gapopen", "qstart", "qend", "sstart", "send", "evalue", "bitscore"))
head(kratom_genes)
kratom_genes$Ath <- gsub("\\.\\d+$", "", kratom_genes$Ath) #remove transcript syntax from arabidopsis genes

leaf_markers <- markers %>% 
  left_join(kratom_genes, by = "Ath") %>% #merge dionne's marker genes and kratom blast hits by arabidopsis genes
  mutate(gene_ID = str_replace(gene_ID, "\\.\\d+$", "")) %>%
  group_by(gene_ID) %>%
  arrange(bitscore, .by_group = TRUE) %>% #arrange multiple hits to the same arabidopsis gene by bitscore
  distinct(gene_ID, .keep_all = T) %>% #keep top hit with highest bitscore in multiple 
  #distinct(Ath, .keep_all = T) %>%
  left_join(anno, by = "gene_ID", multiple = "first") %>%
  select(c(1:5, 17, 6:9, 15:16)) %>%
  ungroup() %>%
  mutate(gene_ID = str_replace(gene_ID, "_", "-"))

#write.table(leaf_markers, file = "~/nih/marker_genes/mitr_v1_markers_04feb25.txt", col.names = T, quote = F)  

#leaf_markers$gene_ID <- gsub(pattern = "_", replacement = "-", leaf_markers$gene_ID)
#leaf_markers$gene_ID <- gsub("\\.\\d+$", "", leaf_markers$gene_ID)

dim(leaf_markers) #176
head(leaf_markers)
```

## Cell type assignment heatmap
```{r}
DefaultAssay(leaf.combined) <- "RNA"

leaf_markers2 <- leaf_markers %>% 
  dplyr::select(symbol, cell_type, gene_ID) %>%
  dplyr::filter(str_detect(gene_ID, "Mitsp.v1.19-1G001940") == F) %>%
  dplyr::filter(str_detect(gene_ID, "Mitsp.v1.08-1G000720") == F) %>% #guard cell that was expressed with phloem cells
  dplyr::filter(str_detect(gene_ID, "Mitsp.v1.20-2G008400") == F) %>% #under-expressed xylem marker
  dplyr::filter(str_detect(gene_ID, "Mitsp.v1.06-1G021870") == F) %>% #under-expressed xylem marker cluster 19
  dplyr::filter(str_detect(gene_ID, "Mitsp.v1.12-1G019890") == F) %>% # redundant xylem marker
  dplyr::filter(str_detect(gene_ID, "Mitsp.v1.16-2G012920") == F) %>% #under-expressed phloem marker
  dplyr::filter(str_detect(gene_ID, "Mitsp.v1.03-1G024630") == F) %>% #under-expressed shoot meristem marker
  dplyr::filter(str_detect(gene_ID, "Mitsp.v1.10-2G011550") == F) %>% #under-expressed proliferating marker
  dplyr::filter(str_detect(gene_ID, "Mitsp.v1.01-2G008870") == F) %>% #under-expressed G0/G1 phase marker
  dplyr::filter(is.na(gene_ID) == F)
  
Marker_dotplot <- DotPlot(leaf.combined, 
                          features = unique(leaf_markers2$gene_ID)) 
head(Marker_dotplot$data)

Marker_dotplot_nice_df <- Marker_dotplot$data %>% 
  mutate(gene_ID = row.names(.)) %>% 
  mutate(gene_ID = str_sub(gene_ID, end = 15)) %>% 
  full_join(leaf_markers2 %>% 
              distinct(gene_ID, .keep_all = T),
             by = c("features.plot" = "gene_ID")) %>% 
   mutate(cell_type2 = case_when(
    str_detect(cell_type, "meso") ~ "M",
    str_detect(cell_type, "cam|x|phl|com|vas") ~ "V",
    str_detect(cell_type, "epi") ~ "E",
    str_detect(cell_type, "guard") ~ "GC",
    str_detect(cell_type, "IPAP") ~ "IP",
    str_detect(cell_type, "idioblast")  ~ "I",
    str_detect(cell_type, "G0|G1|G2|prolif")  ~ "Pro",
    str_detect(cell_type, "shoot") ~ "SAM",
    str_detect(cell_type, "trich") ~ "TRI"
  )) %>% 
  dplyr::filter(str_detect(cell_type, "shea") == F) %>% 
  mutate(cell_type2 = factor(cell_type2, levels = c(
    "M", "E", "GC", "V", "IP", "I", "Pro", "SAM", "TRI"
  ))) %>% 
  mutate(cell_type3 = as.numeric(cell_type2)) %>% 
  mutate(features.plot = reorder(features.plot, cell_type3, decreasing = T)) %>% 
  dplyr::filter(is.na(avg.exp.scaled) == F)%>%
  mutate(tag = paste(features.plot, cell_type2)) %>%
  mutate(id = factor(id, levels = c(
    "0", "2", "3", "7", "8", "11", "15", "17", "22",
    "1", "6", "10", "14", "16",
    "9","21", "12", "13", "18",
    "4", "5","19", "20"
    )))

dim(Marker_dotplot_nice_df) #552
head(Marker_dotplot_nice_df)
```

* Mesophyll: "0", "2", "3", "7", "8", "11", "15", "17", "22",
* Epi: "1", "6", "10", "14", "16",
* Guard: 
* Vasculature: "9","21", "12", "13", "18",
* Un: "4", "5", "7", "15", "19", "20"

```{r}
Marker_dot <- Marker_dotplot_nice_df %>% 
  ggplot(aes(x = id, y = reorder(tag, cell_type3, decreasing = TRUE))) +
  geom_point(aes(size = pct.exp, fill = avg.exp.scaled), shape = 21, color = "white") +
  scale_fill_viridis(option = "A", begin = 0, end = 0.9) +
  guides(size = guide_legend(override.aes = list(fill = "grey40"), nrow = 2)) +
  labs(x = "Cluster",
       y = NULL, 
       fill = "Avg. Exp.",
       size = "% Exp.") +
  theme_classic() +
  theme(text = element_text(size = 14, color = "black"),
        axis.text = element_text(color = "black"),
        legend.position = "top",
        legend.box = "horizontal",
        legend.key.height = unit(0.7, "lines"))

Marker_dot
ggsave("/scratch/ac05869/nih/kratom/unordered_MIT_AH_AI_clusters_with_Dionne_deduplicated_markers_04feb25.png", width = 10, height = 8)
```

#Merge seurat cluster top marker file with leaf_markers to identify top markers for each cluster
```{r}
cluster_markers <- read_delim("/scratch/ac05869/nih/kratom_pathway/MIT_AH_AI_clean_integrated_all_markers_minpct0.25_minlfc0.25_with_info_30jan25.txt", delim = "\t", col_names = T) %>% rename("gene_ID" = "ID")
cluster_markers$gene_ID <- gsub(pattern = "_", replacement = "-", cluster_markers$gene_ID)
head(cluster_markers)
head(leaf_markers)

cluster_markers2 <- inner_join(cluster_markers, leaf_markers, by = "gene_ID") %>% 
  group_by(cluster) %>%
  select(1, 11, 2, 13, 12, 8, 9, 10, 14, 17, 21)
head(cluster_markers2)

#remove commas from Reference column in order to write csv file
cluster_markers2 <- cluster_markers2 %>%
  mutate(Reference = str_replace(Reference, ",", " ")) %>%
           write_csv(file = "~/nih/marker_genes/merged_Dionne_markers_MIT_AH_AI_all_markers_minpct0.25_minlfc0.25_05feb25.csv", quote = "none")
head(cluster_markers2)
```
* Mesophyll: "0", "2", "3", "7", "8", "11", "15", "17", "22",
* Epi: "1", "6", "10", "14", "16",
* Guard: 
* Vasculature: "9","21", "12", "13", "18",
* Un: "4", "5", "19", "20"

```{r}
leaf_cell_type_assignment <- data.frame(
  cluster = c(0:22)
) %>% 
  mutate(cell_type = case_when(
    cluster == 0 |
      cluster == 2 |
      cluster == 3 |
      cluster == 7 |
      cluster == 8 | 
      cluster == 11 |
      cluster == 15 |
      cluster == 17 |
      cluster == 22  ~ "Mesophyll",
    cluster == 1 |
      cluster == 6 |
      cluster == 10 |
      cluster == 14 |
      cluster == 16 ~ "Epidermis",
    cluster == 9 |
      cluster == 21 |
      cluster == 12 |
      cluster == 13 |
      cluster == 18  ~ "Vasculature",
    cluster == 4 |
      cluster ==5 |
      cluster ==19 |
      cluster == 20 ~ "Unassigned"
  )) %>% 
  mutate(cell_type = factor(cell_type, 
                            levels = c(
                              "Mesophyll", "Epidermis", "Guard cells",
                              "Vasculature", 
                              "Unassigned"
                            ))) %>% 
  mutate(cluster = factor(cluster, levels = c(
     "0", "2", "3", "7", "8", "11", "15", "17", "22",
     "1", "6", "10", "14", "16",
     "9","21", "12", "13", "18",
     "4", "5", "19", "20"
    )))
```


```{r}
Leaf_cell_type_strip <- leaf_cell_type_assignment %>% 
  ggplot(aes(x = cluster, y = "" )) +
  geom_tile(aes(fill = cell_type)) +
  scale_fill_manual(values = c(brewer.pal(6, "Accent"), "grey80")) +
  guides(fill = guide_legend(nrow = 3)) +
  labs(fill = "Cell type") +
  theme_void() +
  theme(
    legend.position = "bottom" ,
    text = element_text(size = 14)
  )

Leaf_cell_type_strip

Leaf_cell_type_mkr_text <- Marker_dotplot_nice_df %>% 
  dplyr::filter(id == 0) %>% 
  ggplot(aes(x = "", y = reorder(tag, cell_type3, decreasing = TRUE))) +
  geom_text(aes(label = unique(symbol), fontface = "italic")) +
  theme_void()

blank <- data.frame(
  x = 0,
  y = 0
) %>% 
  ggplot(aes(x, y)) +
  theme_void()

Marker_dot_strip <- wrap_plots(Marker_dot, Leaf_cell_type_mkr_text,
          Leaf_cell_type_strip, blank,
          heights = c(1, 0.03), 
          widths = c(1, 0.2), nrow = 2, ncol = 2)

Marker_dot_strip

ggsave("/scratch/ac05869/nih/kratom/MIT_AH_AI_cell_type_assignment_plot.svg", height = 8, width = 7.5, bg = "white")
ggsave("/scratch/ac05869/nih/kratom/MIT_AH_AI_cell_type_assignment_plot.png", height = 6, width = 7.5, bg = "white")
```


