---
title: "Integrate UGA and MPI single cell analysis with mitr_v1.asm"
author: "Anne Pavia"
output:
  html_document:
    df_print: paged
#Notes: 
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
#knitr::opts_knit$set(root.dir = "/scratch/ac05869/MIT_KRT_sc_analysis/Rwd")
library(tidyverse)
library(ggplot2)
library(Seurat) 
library(readxl)
library(RColorBrewer)
library(viridis)
library(svglite) 
library(patchwork)
library(cowplot)
library(ggpubr)
library(ape)
library(igraph)
library(showtext)
#showtext_auto()
```
Notes: AA and AB are reps of KRT uga leaf single cell data. AH and AI are reps of mpi leaf single cell data.


# Load RDS Object post doublet filtering
```{r}
AA_clean <- readRDS("/scratch/ac05869/MIT_KRT_sc_analysis/seurat_data/KRT_AA_doublets_removed_21feb25.rds")
AB_clean <- readRDS("/scratch/ac05869/MIT_KRT_sc_analysis/seurat_data/KRT_AB_doublets_removed_21feb25.rds")
AH_clean <- readRDS("/scratch/ac05869/MIT_KRT_sc_analysis/seurat_data/MIT_AH_doublets_removed_21feb25.rds")
AI_clean <- readRDS("/scratch/ac05869/MIT_KRT_sc_analysis/seurat_data/MIT_AI_doublets_removed_21feb25.rds")

AA_clean$source <- "UGA"
AB_clean$source <- "UGA"
AH_clean$source <- "MPI"
AI_clean$source <- "MPI"
```

# Re-Clustering Cleaned Data 

Now that we have a clean dataset we need to re-cluster things to account for the removal of doublets 
We can either re-run the above clustering steps for single samples OR we can integrate our replicates together at the same time

## Normalize the Data
```{r}

#Default = LogNormalize

AA_clean <- NormalizeData(AA_clean)
AB_clean <- NormalizeData(AB_clean)
AH_clean <- NormalizeData(AH_clean)
AI_clean <- NormalizeData(AI_clean)
```

# Integrate all 4 reps at once
## Integration via reciprocal PCA (RPCA)
###Data Prep

```{r}
leaf.list <- list(AA_clean, AB_clean, AH_clean, AI_clean)

# normalize and identify variable features for each dataset independently
leaf.list <- lapply(X = leaf.list, FUN = function(x) {
    x <- NormalizeData(x)
    x <- FindVariableFeatures(x, selection.method = "vst", nfeatures = 3000)
})

# select features that are repeatedly variable across datasets for integration run PCA on each dataset using these features
features <- SelectIntegrationFeatures(object.list = leaf.list, nfeatures = 3000)
leaf.list <- lapply(X = leaf.list, FUN = function(x) {
    x <- ScaleData(x, features = features, verbose = FALSE)
    x <- RunPCA(x, features = features, verbose = FALSE)
})
```

### Integration 
```{r}
leaf.anchors <- FindIntegrationAnchors(object.list = leaf.list, anchor.features = features, reduction = "rpca")

# this command creates an 'integrated' data assay
leaf.combined <- IntegrateData(anchorset = leaf.anchors)

# specify that we will perform downstream analysis on the corrected data note that the
# original unmodified data still resides in the 'RNA' assay
DefaultAssay(leaf.combined) <- "integrated"
```

## Stats
```{r}
leaf.combined

summary(leaf.combined$nFeature_RNA)
summary(leaf.combined$nCount_RNA)

```

## Scale the Data
```{r}
leaf.combined.genes <- rownames(leaf.combined)
leaf.combined <- ScaleData(leaf.combined, features = leaf.combined.genes)
```

## PCA
```{r}
leaf.combined$Run <- leaf.combined@meta.data$orig.ident
leaf.combined <- RunPCA(leaf.combined, npcs = 75, features = VariableFeatures(object = leaf.combined), verbose = FALSE) # run PCA
```

```{r}
ElbowPlot(leaf.combined, ndims = 75)
```

## Call Clusters
```{r}
leaf.combined <- FindNeighbors(leaf.combined, dims = 1:50) #adjust dims based on elbow plot above
leaf.combined <- FindClusters(leaf.combined, resolution = 0.4) #adjustable resolution value. Check back and forth until it looks good
leaf.combined <- RunUMAP(leaf.combined, dims = 1:50)
```

## Make UMAP Plots

### By Run

```{r}
UMAP1 <- DimPlot(leaf.combined, reduction = "umap", label = T, label.size = 12, repel = T, split.by = "Run") +
  theme_void() +
  theme(
    text = element_text(size = 26, color = "black", face = "bold"),
    legend.position = "none",
    title = element_text(size = 20)
  ) +
  ggtitle("Grouped by Replicates\n")

UMAP1 #check to see that there are the same number of cluster in each rep: KRT_AB does not have cluster 23

ggsave("../seurat_out/KRT_MIT_leaf_integrated_umap_by_run.png", height = 6, width = 18, bg = "white")
```
### By source

```{r}
UMAP2 <- DimPlot(leaf.combined, reduction = "umap", label = T, label.size = 12, repel = T, split.by = "source") +
  theme_void() +
  theme(
    text = element_text(size = 26, color = "black", face = "bold"),
    legend.position = "none",
    title = element_text(size = 20)
  ) +
  ggtitle("Grouped by Source")

UMAP2 #check to see that there are the same number of cluster in each rep: KRT_AB does not have cluster 23

ggsave("../seurat_out/KRT_MIT_leaf_integrated_umap_by_source.png", height = 6, width = 18, bg = "white")
```



### Combined
```{r}
UMAP3 <- DimPlot(leaf.combined, reduction = "umap", label = T, label.size = 12, repel = T) +
  theme_void() +
  theme(
    text = element_text(size = 26, color = "black", face = "bold"),
    legend.position = "none",
    title = element_text(size = 20)
  )
UMAP3
ggsave("../seurat_out/KRT_MIT_leaf_integrated_umap_all_together.png", height = 8, width = 12, bg = "white")
```

## Stats - Integrated Clusters
```{r}
#Cells in each cluster
table(Idents(leaf.combined))

# Cells in each replicate
table(leaf.combined$orig.ident)

#How does cluster membership vary by replicate?
table(Idents(leaf.combined), leaf.combined$orig.ident)

#How does cluster membership vary by source?
table(Idents(leaf.combined), leaf.combined$orig.ident)
```

## Save cleaned integrated RDS Object
```{r}
#saveRDS(leaf.combined, file = "/scratch/ac05869/MIT_KRT_sc_analysis/seurat_data/KRT_MIT_cleaned_integrated_seurat_21feb25.rds")

leaf.combined <- readRDS("/scratch/ac05869/MIT_KRT_sc_analysis/seurat_data/KRT_MIT_cleaned_integrated_seurat_21feb25.rds")
```

# Create Matrices
## Find Markers for combined dataset
```{r}
DefaultAssay(leaf.combined) <- "RNA"

leaf.combined = JoinLayers(leaf.combined)
head(leaf.combined)

leaf.combined.markers <- FindAllMarkers(leaf.combined, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25) #min.pct is the minimum fraction that cells cluster needs to be expressing that gene for it to count. logfc.threshhold is the log-fold change in expression between clusters expressing that gene
head(leaf.combined.markers)

leaf.combined.markers %>% group_by(cluster)
#write.table(leaf.combined.markers, "/scratch/ac05869/MIT_KRT_sc_analysis/seurat_out/KRT_MIT_clean_integrated_all_markers_minpct0.25_minlfc0.25_21feb25.txt", quote = FALSE, sep = "\t", row.names = FALSE)
```


```{r}
#saveRDS(leaf.combined.markers, file = "/scratch/ac05869/MIT_KRT_sc_analysis/seurat_out/leaf.combined.markers.rds")
leaf.combined.markers <- readRDS(file = "/scratch/ac05869/MIT_KRT_sc_analysis/seurat_out/leaf.combined.markers.rds")
```

#find markers for each leaf source individually
```{r}
DefaultAssay(leaf.combined) <- "RNA"
leaf.combined.split <- SplitObject(leaf.combined, split.by = "source")
head(leaf.combined.split)
head(leaf.combined.split$MPI)#check that it looks the same as when it is together

leaf.combined.split.MPI = JoinLayers(leaf.combined.split$MPI)
head(leaf.combined.split.MPI)

leaf.combined.split.UGA = JoinLayers(leaf.combined.split$UGA)
head(leaf.combined.split.UGA)

leaf.combined.markers.MPI <- FindAllMarkers(leaf.combined.split.MPI, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
head(leaf.combined.markers.MPI)

leaf.combined.markers.UGA <- FindAllMarkers(leaf.combined.split.UGA, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
head(leaf.combined.markers.UGA)

#leaf.combined.markers.MPI %>% group_by(cluster) %>% write.table("/scratch/ac05869/MIT_KRT_sc_analysis/seurat_out/KRT_MIT_clean_integrated_MPI_all_markers_minpct0.25_minlfc0.25_28mar25.txt", quote = FALSE, sep = "\t", row.names = FALSE)
#leaf.combined.markers.UGA %>% group_by(cluster) %>% write.table("/scratch/ac05869/MIT_KRT_sc_analysis/seurat_out/KRT_MIT_clean_integrated_UGA_all_markers_minpct0.25_minlfc0.25_28mar25.txt", quote = FALSE, sep = "\t", row.names = FALSE)
```


```{r}
#saveRDS(leaf.combined.markers.MPI, file = "/scratch/ac05869/MIT_KRT_sc_analysis/seurat_out/leaf.combined.markers.MPI.rds")
leaf.combined.markers.MPI <- readRDS(file = "/scratch/ac05869/MIT_KRT_sc_analysis/seurat_out/leaf.combined.markers.MPI.rds")

#saveRDS(leaf.combined.markers.UGA, file = "/scratch/ac05869/MIT_KRT_sc_analysis/seurat_out/leaf.combined.markers.UGA.rds")
leaf.combined.markers.UGA <- readRDS(file = "/scratch/ac05869/MIT_KRT_sc_analysis/seurat_out/leaf.combined.markers.UGA.rds")
```

## Find Average Expression
```{r}
DefaultAssay(leaf.combined) <- "RNA"

# replace spaces with underscores '_' so ggplot2 doesn't fail
orig.levels <- levels(leaf.combined)
Idents(leaf.combined) <- gsub(pattern = " ", replacement = "_", x = Idents(leaf.combined))
orig.levels <- gsub(pattern = " ", replacement = "_", x = orig.levels)
levels(leaf.combined) <- orig.levels
cluster.averages <- AverageExpression(leaf.combined, return.seurat = TRUE)
cluster.averages

# How can I extract expression matrix for all Cluster 0 cells (perhaps, to load into another package)
cluster_avg.data <- as.matrix(GetAssayData(cluster.averages, layer = "data")[, ])
head(cluster_avg.data)
#write.table(cluster_avg.data, file = "/scratch/ac05869/MIT_KRT_sc_analysis/seurat_out/KRT_MIT_clean_integrated_cluster_avg_normalized_counts_21jan25.txt", sep = "\t", quote = FALSE)
saveRDS(cluster_avg.data, file = "/scratch/ac05869/MIT_KRT_sc_analysis/seurat_out/cluster_avg.data.rds")
#cluster_avg.data <- readRDS(file = "/scratch/ac05869/MIT_KRT_sc_analysis/seurat_out/cluster_avg.data.rds")

```

## Find Average Expression for individual UGA and MPI sources
```{r}
#MPI leaf only
DefaultAssay(leaf.combined.split.MPI) <- "RNA"
head(leaf.combined.split.MPI)
# replace spaces with underscores '_' so ggplot2 doesn't fail
(orig.levels <- levels(leaf.combined.split.MPI))
Idents(leaf.combined.split.MPI) <- gsub(pattern = " ", replacement = "_", x = Idents(leaf.combined.split.MPI))
orig.levels <- gsub(pattern = " ", replacement = "_", x = orig.levels)
levels(leaf.combined.split.MPI) <- orig.levels
cluster.averages.MPI <- AverageExpression(leaf.combined.split.MPI, group.by = c("ident", "source"), return.seurat = TRUE)
cluster.averages.MPI

# How can I extract expression matrix for all Cluster 0 cells (perhaps, to load into another package)
cluster_avg_MPI.data <- as.matrix(GetAssayData(cluster.averages.MPI, layer = "data")[, ])
head(cluster_avg_MPI.data)

###Now UGA leaf
DefaultAssay(leaf.combined.split.UGA) <- "RNA"
head(leaf.combined.split.UGA)
# replace spaces with underscores '_' so ggplot2 doesn't fail
(orig.levels <- levels(leaf.combined.split.UGA))
Idents(leaf.combined.split.UGA) <- gsub(pattern = " ", replacement = "_", x = Idents(leaf.combined.split.UGA))
orig.levels <- gsub(pattern = " ", replacement = "_", x = orig.levels)
levels(leaf.combined.split.UGA) <- orig.levels
cluster.averages.UGA <- AverageExpression(leaf.combined.split.UGA, group.by = c("ident", "source"), return.seurat = TRUE)
cluster.averages.UGA

# How can I extract expression matrix for all Cluster 0 cells (perhaps, to load into another package)
cluster_avg_UGA.data <- as.matrix(GetAssayData(cluster.averages.UGA, layer = "data")[, ])
head(cluster_avg_UGA.data)

#Write tables
#write.table(cluster_avg_MPI.data, file = "/scratch/ac05869/MIT_KRT_sc_analysis/seurat_out/KRT_MIT_clean_integrated_MPI_cluster_avg_normalized_counts_28mar25.txt", sep = "\t", quote = FALSE, row.names = T)
#write.table(cluster_avg_UGA.data, file = "/scratch/ac05869/MIT_KRT_sc_analysis/seurat_out/KRT_MIT_clean_integrated_UGA_cluster_avg_normalized_counts_28mar25.txt", sep = "\t", quote = FALSE, row.names = T)
```


```{r}
#saveRDS(cluster_avg_MPI.data, file = "/scratch/ac05869/MIT_KRT_sc_analysis/seurat_out/cluster_avg_MPI.data.rds")
cluster_avg_MPI.data <- readRDS(file = "/scratch/ac05869/MIT_KRT_sc_analysis/seurat_out/cluster_avg_MPI.data.rds")

#saveRDS(cluster_avg_UGA.data, file = "/scratch/ac05869/MIT_KRT_sc_analysis/seurat_out/cluster_avg_UGA.data.rds")
cluster_avg_UGA.data <- readRDS(file = "/scratch/ac05869/MIT_KRT_sc_analysis/seurat_out/cluster_avg_UGA.data.rds")
```

#Annotate marker lists
## Read in annotation file and working models gff
```{r}
anno <- read.delim(file = "/scratch/ac05869/nih/kratom/mitr_v1_anno/mitr_v1.functional_annotation.txt", header = FALSE, sep = "\t", col.names = c("gene", "functional_annotation"))
#Don't remove transcript syntax because I want to select only the representative transcript
#anno$gene <- gsub("\\.\\d+$", "", anno$gene)
head(anno)
nrow(anno) #172819

##Read in repr gff file
working_models <- read.gff("/work/crblab/ac05869/genomes/mitr_v1.working_models.repr.gff3", na.strings = c(".", "?"), GFF3 = TRUE)
head(working_models)
#parse apart attributes column
head(attributes <- working_models$attributes)
head(parsed_attributes <- strsplit(attributes, ";"))
extract_attribute_value <- function(attribute_list, key) {
  key_value <- grep(paste0("^", key, "="), attribute_list, value = TRUE)
  if (length(key_value) > 0) {
    return(sub(paste0("^", key, "="), "", key_value))
  } else {
    return(NA)
  }
}
working_models$Name <- sapply(parsed_attributes, extract_attribute_value, key = "Name")
working_models$ID <- sapply(parsed_attributes, extract_attribute_value, key = "ID")
working_models$Parent <- sapply(parsed_attributes, extract_attribute_value, key = "Parent")

working_models2 <- working_models %>%
  select(c(ID, seqid, type, start, end, strand)) %>% 
  filter(type == "mRNA") %>%
  rename(chr = seqid) %>%
  rename(gene = ID)
#working_models2$gene <- gsub("\\.\\d+$", "", working_models2$gene) #Don't remove transcript syntax yet. Need to select repr transcript in annotation file
head(working_models2)
dim(working_models2) #103160
```




## Annotate marker list
```{r}
#replace "-" with underscores '_' so that it matches the gff file
leaf.combined.markers$gene <- gsub(pattern = "-", replacement = "_", leaf.combined.markers$gene)
head(leaf.combined.markers)

leaf.combined.markers.MPI$gene <- gsub(pattern = "-", replacement = "_", leaf.combined.markers.MPI$gene)
head(leaf.combined.markers.MPI)

leaf.combined.markers.UGA$gene <- gsub(pattern = "-", replacement = "_", leaf.combined.markers.UGA$gene)
head(leaf.combined.markers.UGA)

#Need to add chr, pos_left, pos_right, best_arab_hit, best_arab_hit_functional_annotation, CoExpression_Module_Membership, avg_log2FC

#UGA and MPI combined list
combined_markers_with_gff_anno <- leaf.combined.markers %>%
  inner_join(working_models2, by="gene") %>%
  inner_join(anno, by = "gene") %>%
  select(cluster, gene, functional_annotation, avg_log2FC, p_val_adj, pct.1, pct.2, chr, start, end, strand)
combined_markers_with_gff_anno$gene <- gsub("\\.\\d+$", "", combined_markers_with_gff_anno$gene) #Now remove syntax to match seurat
dim(combined_markers_with_gff_anno) #103160
head(combined_markers_with_gff_anno)
write.table(combined_markers_with_gff_anno, file = "/scratch/ac05869/MIT_KRT_sc_analysis/seurat_out/annotated_KRT_MIT_clean_integrated_all_markers_minpct0.25_minlfc0.25_4apr25.txt", row.names = FALSE, sep = "\t", quote = FALSE)

#Split MPI markers only
MPI_markers_with_gff_anno <- inner_join(leaf.combined.markers.MPI, working_models2, by="gene") %>%
  left_join(anno, by = "gene") %>%
  select(cluster, gene, functional_annotation, avg_log2FC, p_val_adj, pct.1, pct.2, chr, start, end, strand)
dim(MPI_markers_with_gff_anno) #34029
head(MPI_markers_with_gff_anno)
write.table(MPI_markers_with_gff_anno, file = "/scratch/ac05869/MIT_KRT_sc_analysis/seurat_out/annotated_KRT_MIT_clean_integrated_MPI_markers_minpct0.25_minlfc0.25_4apr25.txt", row.names = FALSE, sep = "\t", quote = FALSE)

#Split UGA markers only
UGA_markers_with_gff_anno <- inner_join(leaf.combined.markers.UGA, working_models2, by="gene") %>%
  left_join(anno, by = "gene") %>%
  select(cluster, gene, functional_annotation, avg_log2FC, p_val_adj, pct.1, pct.2, chr, start, end, strand)
dim(UGA_markers_with_gff_anno) #34029
head(UGA_markers_with_gff_anno)
write.table(UGA_markers_with_gff_anno, file = "/scratch/ac05869/MIT_KRT_sc_analysis/seurat_out/annotated_KRT_MIT_clean_integrated_UGA_markers_minpct0.25_minlfc0.25_4apr25.txt", row.names = FALSE, sep = "\t", quote = FALSE)
```
# Annotate with Coexpression data
## Read in coexpression table of modules
```{r}
#GOI
pathway_genes <- read_delim("/scratch/ac05869/MIT_KRT_sc_analysis/kratom_pathway_orthologs_24mar25.txt", delim = "\t", col_names = T, col_types = cols())
pathway_genes <- pathway_genes %>%
  rename(gene = kratom.id)
head(pathway_genes)

#Module information
Exp_table_long_averaged_z_modules <- read_delim(file = "/scratch/ac05869/MIT_KRT_sc_analysis/coexp_out/Exp_table_long_averaged_z_modules_r0.7_3res.csv", delim = ",") %>%
  select(gene_ID, cluster, avg_exp, z.score, module, cell_type) %>%
  rename(gene = gene_ID) %>%
  rename(module_avg_exp = avg_exp)%>%
  mutate(cluster = as.factor(cluster))
Exp_table_long_averaged_z_modules$gene <- gsub(pattern = "-", replacement = "_", Exp_table_long_averaged_z_modules$gene)
head(Exp_table_long_averaged_z_modules)

#Network information
edge_table_select <- readRDS("/scratch/ac05869/MIT_KRT_sc_analysis/coexp_out/KRT_MIT_leaf_edge_table_select_r.7.rds")
#edge_table_select$from <- gsub(pattern = "-", replacement = "_", edge_table_select$from)
#edge_table_select$from <- gsub(pattern = "-", replacement = "_", edge_table_select$to)
head(edge_table_select)
dim(edge_table_select) #36467739

node_table <- data.frame(
  gene_ID = c(edge_table_select$from, edge_table_select$to) %>% 
    unique())
head(node_table)
dim(node_table) #60851
  
my_network <- graph_from_data_frame(
  edge_table_select,
  vertices = node_table,
  directed = F
)

#head(pathway_genes)
#baits <- list(pathway_genes$gene)
#baits

neighbors1 <- neighbors(my_network, v = "Mitsp.v1.18-2G016780")
length(neighbors1)

neighbors_of_bait <- c(
  neighbors(my_network, v = "Mitsp.v1.18-2G016780"), #MsSAS 
  neighbors(my_network, v = "Mitsp.v1.10-1G010920"), #MsDCR1
  neighbors(my_network, v = "Mitsp.v1.10-2G012340"), #MsDCR2
  neighbors(my_network, v = "Mitsp.v1.08-2G007110"), #Ms10H
  neighbors(my_network, v = "Mitsp.v1.10-2G022130"),# ~ "MsCO1.1",
  neighbors(my_network, v = "Mitsp.v1.10-1G021390"), # ~ "MsCO1.2",
  neighbors(my_network, v = "Mitsp.v1.10-1G021550"), # ~ "MsCO1.3",
  neighbors(my_network, v = "Mitsp.v1.10-1G015530"), # ~ "MsDCS1",
  neighbors(my_network, v = "Mitsp.v1.10-1G015520"), # ~ "MsDCS2",
  neighbors(my_network, v = "Mitsp.v1.02-1G013220"), # ~ "MsSGD",
  neighbors(my_network, v = "Mitsp.v1.11-1G000250")) %>% #MsEnolMT
  unique()
length(neighbors_of_bait) #6698

subnetwork_edges <- edge_table_select %>% 
  filter(from %in% names(neighbors_of_bait) &
           to %in% names(neighbors_of_bait)) %>% 
  group_by(from) %>% 
  slice_max(order_by = r, n = 5) %>% 
  ungroup() %>% 
  group_by(to) %>% 
  slice_max(order_by = r, n = 5) %>% 
  ungroup()

subnetwork_genes <- c(subnetwork_edges$from, subnetwork_edges$to) %>% unique()
length(subnetwork_genes) #6660
#View(subnetwork_genes)
dim(subnetwork_edges) #17142
#View(subnetwork_edges)

subnetwork_edges$from <- as.factor(subnetwork_edges$from)
subnetwork_edges$from <- gsub("-", "_", subnetwork_edges$from)
subnetwork_edges$to <- gsub("-", "_", subnetwork_edges$to)

subnetwork_neighbors_df <- subnetwork_edges %>%
  group_by(from) %>%
  summarise(neighbors = paste(to, collapse = ",")) %>%
  rename(gene = from)
  
head(subnetwork_neighbors_df)
```
We can constrain the edges such that both the start and end of edges are neighbors of baits. 
I also filtered for highly correlated neighbors (top 5 edges/node based on r value). 

```{r}
#integrated mpi and uga markers
combined_markers_with_gff_anno_modules <- combined_markers_with_gff_anno %>%
  group_by(cluster) #%>%
  left_join(Exp_table_long_averaged_z_modules, by = c("gene" = "gene", "cluster" = "cluster")) %>%
  left_join(pathway_genes, by = "gene") %>%
  left_join(subnetwork_neighbors_df, by = "gene")
combined_markers_with_gff_anno_modules

write.table(combined_markers_with_gff_anno_modules, file = "/scratch/ac05869/MIT_KRT_sc_analysis/seurat_out/annotated_coexpression_KRT_MIT_clean_integrated_all_markers_minpct0.25_minlfc0.25_4apr25.txt", row.names = FALSE, sep = "\t", quote = FALSE)


# mpi markers
MPI_markers_with_gff_anno_modules <- MPI_markers_with_gff_anno %>%
  group_by(cluster) %>%
  left_join(Exp_table_long_averaged_z_modules, by = c("gene" = "gene", "cluster" = "cluster")) %>%
  left_join(pathway_genes, by = "gene")%>%
  left_join(subnetwork_neighbors_df, by = "gene")
MPI_markers_with_gff_anno_modules

write.table(MPI_markers_with_gff_anno_modules, file = "/scratch/ac05869/MIT_KRT_sc_analysis/seurat_out/annotated_coexpression_KRT_MIT_clean_integrated_MPI_markers_minpct0.25_minlfc0.25_4apr25.txt", row.names = FALSE, sep = "\t", quote = FALSE)


#uga markers
UGA_markers_with_gff_anno_modules <- UGA_markers_with_gff_anno %>%
  group_by(cluster) %>%
  left_join(Exp_table_long_averaged_z_modules, by = c("gene" = "gene", "cluster" = "cluster")) %>%
  left_join(pathway_genes, by = "gene")%>%
  left_join(subnetwork_neighbors_df, by = "gene")
MPI_markers_with_gff_anno_modules

write.table(MPI_markers_with_gff_anno_modules, file = "/scratch/ac05869/MIT_KRT_sc_analysis/seurat_out/annotated_coexpression_KRT_MIT_clean_integrated_UGA_markers_minpct0.25_minlfc0.25_4apr25.txt", row.names = FALSE, sep = "\t", quote = FALSE)
```

# Make Plots
## Look at top several pathway genes
```{r}
#Need to create a file that has your marker genes and some metadata (cell types, etc.). Change all underscores to either dashes or periods

allwin <- read.table("/scratch/ac05869/MIT_KRT_sc_analysis/pathway_data/allwin_pathay_top2_25feb25.txt", header = TRUE, sep = "\t")
allwin$kratom.id <- gsub("\\.\\d+$", "", allwin$kratom.id) #remove transcript syntax to match seurat object
allwin$kratom.id <- gsub("_", "-", allwin$kratom.id)
allwin <- allwin %>% distinct(kratom.id, .keep_all = T)
  #remove duplicates
head(allwin)

rev_allwin <- allwin[order(nrow(allwin):1),]
```

## Find missing genes
```{r}
DefaultAssay(leaf.combined) <- "RNA"

DotPlot(leaf.combined, features = unique(rev_allwin$kratom.id), 
        ) +
  scale_x_discrete(labels=rev_allwin$gene.abbreviation) +
  scale_color_gradientn(colors = viridis(10, option = "A")[1:9]) +
  labs(y = "Cluster",
       x = "Genes") + 
  coord_flip()
```

## Remove genes
```{r}
#Remove everything that has no expression based on the error messages from making the above plot
#View(allwin) ##see row in which the variable was not found in order to subset
subset <- str_trim(paste(scan("/scratch/ac05869/MIT_KRT_sc_analysis/pathway_data/missing_in_leaf.txt", what = "character", sep = c(",", " "))))

allwin_subset <- allwin %>%
  filter(! as.character(kratom.id) %in% as.character(subset))

(rev_allwin_subset <- allwin_subset %>% arrange(desc(order))) #another way to do it if your data has "order" column
```

## Make dot plot
```{r}
showtext_auto(enable = FALSE)
#plot only genes with expression
(dotplot_data <- DotPlot(leaf.combined, features = unique(rev_allwin_subset$kratom.id))$data)
(unique(rev_allwin_subset$kratom.id)) #Count if any duplicate gene ID's

dotplot_data %>% 
  inner_join(rev_allwin_subset %>% 
               distinct(kratom.id, .keep_all = T), by = c("features.plot"="kratom.id")) %>%
  mutate(tag = paste(gene.abbreviation, features.plot)) %>%
  ggplot(aes(x = id, y = reorder(tag,order, decreasing = TRUE))) +
  geom_point(aes(size = pct.exp, color = avg.exp.scaled)) +
  scale_color_gradientn(colors = viridis(10, option = "A")[1:9]) +
  labs(y = "Genes",
       x = "Cluster") + 
  theme_classic()

ggsave("/scratch/ac05869/MIT_KRT_sc_analysis/pathway_data/KRT_MIT_allwin_pathway_top2_hits_25feb25.png", height = 10, width = 12, bg = "white")
```

## Make dot plot split by source (MPI vs UGA)
```{r}
showtext_auto()
#plot only genes with expression
dotplot_data <- DotPlot(leaf.combined, features = unique(rev_allwin_subset$kratom.id), split.by = "source")$data %>%
  mutate(source = case_when(
    str_detect(id, "UGA") ~ "UGA",
    str_detect(id, "MPI") ~ "MPI"
  )) %>%
  mutate(id2 = str_sub(id, start = 1, end = -5))
unique(rev_allwin_subset$kratom.id) #Count if any duplicate gene ID's

dotplot_data %>% 
  inner_join(rev_allwin_subset %>% 
               distinct(kratom.id, .keep_all = T), by = c("features.plot"="kratom.id")) %>%
  mutate(tag = paste(gene.abbreviation, features.plot)) %>%
  ggplot(aes(x = id2, y = reorder(tag,order, decreasing = TRUE))) +
  facet_wrap(~ source) +
  geom_point(aes(size = pct.exp, color = avg.exp.scaled)) +
  theme(text = element_text(size = 26)) +
  scale_color_gradientn(colors = viridis(10, option = "A")[1:9]) +
  labs(y = "Genes",
       x = "Cluster") + 
  theme_classic()

dotplot_data

ggsave("/scratch/ac05869/MIT_KRT_sc_analysis/pathway_data/by_source_KRT_MIT_allwin_pathway_top2_hits_25feb25.png", height = 10, width = 12, bg = "white")
```

## Look at top hit
```{r}
#Need to create a file that has your marker genes and some metadata (cell types, etc.). Change all underscores to either dashes or periods

allwin <- read.table("/scratch/ac05869/MIT_KRT_sc_analysis/pathway_data/allwin_top_pathway_gene_leaf.txt", header = TRUE, sep = "\t")
allwin$kratom.id <- gsub("\\.\\d+$", "", allwin$kratom.id) #remove transcript syntax to match seurat object
allwin$kratom.id <- gsub("_", "-", allwin$kratom.id)
allwin <- allwin %>%
  distinct(kratom.id, .keep_all = T) #remove duplicates
head(allwin)

rev_allwin <- allwin[order(nrow(allwin):1),]
```

## Find missing genes
```{r}
DefaultAssay(leaf.combined) <- "RNA"

DotPlot(leaf.combined, features = unique(rev_allwin$kratom.id), 
        ) +
  scale_x_discrete(labels=rev_allwin$gene.abbreviation) +
  scale_color_gradientn(colors = viridis(10, option = "A")[1:9]) +
  labs(y = "Cluster",
       x = "Genes") + 
  coord_flip()

#missing in top hit: Mitsp.v1.03-2G022350 (TDC), used the 4th best hit for TDC (Mitsp.v1.21_1G011780.1)
```

## Make dot plot
```{r}
#plot only genes with expression
(dotplot_data <- DotPlot(leaf.combined, features = unique(rev_allwin_subset$kratom.id))$data)
(unique(rev_allwin_subset$kratom.id)) #Count if any duplicate gene ID's

dotplot_data %>% 
  inner_join(rev_allwin_subset %>% 
               distinct(kratom.id, .keep_all = T), by = c("features.plot"="kratom.id")) %>%
  mutate(tag = paste(gene.abbreviation, features.plot)) %>%
  ggplot(aes(x = id, y = reorder(tag,order, decreasing = TRUE))) +
  geom_point(aes(size = pct.exp, color = avg.exp.scaled)) +
  theme(text = element_text(size = 26)) +
  scale_color_gradientn(colors = viridis(10, option = "A")[1:9]) +
  labs(y = "Genes",
       x = "Cluster") + 
  theme_classic()

ggsave("/scratch/ac05869/MIT_KRT_sc_analysis/pathway_data/seurat_KRT_MIT_leaf_allwin_top_hit_25feb25.png", height = 10, width = 12, bg = "white")
```

# C. roseus pathway dotplot 
```{r}
cros <- read.table("/scratch/ac05869/MIT_KRT_sc_analysis/cros_allwin_in_kratom.txt", header = TRUE, sep = "\t")
cros$kratom.id <- gsub("\\.\\d+$", "", cros$kratom.id) #remove transcript syntax to match seurat object
cros$kratom.id <- gsub("_", "-", cros$kratom.id)
rev_cros <- cros[order(nrow(cros):1),]
head(rev_cros)
```

## Make dot plot
```{r}
#plot only genes with expression
dotplot_data <- DotPlot(leaf.combined, features = unique(rev_cros$kratom.id))$data

dotplot_data %>% 
  inner_join(rev_cros %>% 
               distinct(kratom.id, .keep_all = T), by = c("features.plot"="kratom.id")) %>%
  mutate(tag = paste(gene.abbreviation, features.plot)) %>%
  ggplot(aes(x = id, y = reorder(tag,order, decreasing = TRUE))) +
  geom_point(aes(size = pct.exp, color = avg.exp.scaled)) +
  theme(text = element_text(size = 26)) +
  scale_color_gradientn(colors = viridis(10, option = "A")[1:9]) +
  labs(y = "Genes",
       x = "Cluster") + 
  theme_classic()

ggsave("/scratch/ac05869/MIT_KRT_sc_analysis/pathway_data/seurat_KRT_MIT_leaf_cros_allwin_pathway_top_hit_25feb25.png", height = 10, width = 12, bg = "white")
```

## Make dot plot split by source (MPI vs UGA)
```{r}
showtext_auto(enable = FALSE)
#plot only genes with expression
dotplot_data <- DotPlot(leaf.combined, features = unique(rev_cros$kratom.id), split.by = "source")$data %>%
  mutate(source = case_when(
    str_detect(id, "UGA") ~ "UGA",
    str_detect(id, "MPI") ~ "MPI"
  )) %>%
  mutate(id2 = str_sub(id, start = 1, end = -5))

dotplot_data %>% 
  inner_join(rev_cros %>% 
               distinct(kratom.id, .keep_all = T), by = c("features.plot"="kratom.id")) %>%
  mutate(tag = paste(gene.abbreviation, features.plot)) %>%
  ggplot(aes(x = id2, y = reorder(tag,order, decreasing = TRUE))) +
  facet_wrap(~ source) +
  geom_point(aes(size = pct.exp, color = avg.exp.scaled)) +
  theme(text = element_text(size = 26)) +
  scale_color_gradientn(colors = viridis(10, option = "A")[1:9]) +
  labs(y = "Genes",
       x = "Cluster") + 
  theme_classic()

dotplot_data

ggsave("/scratch/ac05869/MIT_KRT_sc_analysis/pathway_data/by_source_KRT_MIT_MEP_IR_Allwin_pathway_25feb25.png", height = 10, width = 12, bg = "white")
```

